consensus/src/block_graph.rs

```diff
@@ -51,36 +51,63 @@ impl HeaderOrBlock {
     }
 }
 
+/// Agregated changes made during a block's execution
 #[derive(Debug, Clone)]
 pub struct BlockStateAccumulator {
+    /// Addresses impacted by ledger updates
     pub loaded_ledger_addrs: AddressHashSet,
+    /// Subset of the ledger. Contains only data in the thread of the given block
     pub ledger_thread_subset: LedgerSubset,
+    /// Cummulative changes made during that block execution
     pub ledger_changes: LedgerChanges,
+    /// Addresses impacted by roll updates
     pub loaded_roll_addrs: AddressHashSet,
+    /// Current roll counts for these addresses
     pub roll_counts: RollCounts,
+    /// Roll updates that happened during that block execution
     pub roll_updates: RollUpdates,
+    /// Roll updates that happend during current cycle
     pub cycle_roll_updates: RollUpdates,
+    /// Cycle of the parent in the same thread
     pub same_thread_parent_cycle: u64,
+    /// Address of the parent in the same thread
     pub same_thread_parent_creator: Address,
+    /// Adresses of that block endorsers
     pub endorsers_addresses: Vec<Address>,
 }
 
+/// Block that was checked as final, with some useful precomputed data
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct ActiveBlock {
+    /// The cretor's address
     pub creator_address: Address,
+    /// The block itself, as it was created
     pub block: Block,
-    pub parents: Vec<(BlockId, u64)>, // one (hash, period) per thread ( if not genesis )
-    pub children: Vec<BlockHashMap<u64>>, // one HashMap<hash, period> per thread (blocks that need to be kept)
-    pub dependencies: BlockHashSet,       // dependencies required for validity check
+    /// one (block id, period) per thread ( if not genesis )
+    pub parents: Vec<(BlockId, u64)>,
+    /// one HashMap<Block id, period> per thread (blocks that need to be kept)
+    /// Children reference that block as a parent
+    pub children: Vec<BlockHashMap<u64>>,
+    /// dependencies required for validity check (TODO: not sure if it's used)
```

Comment:
@damip are we using dependencies once the block is active ?

Reply:
yes, for graph pruning, to know which other blocks it depends on